updater/lib/dependabot/job.rb

```diff
@@ -43,124 +45,180 @@ class Job
       dependency_groups
       dependency_group_to_refresh
       repo_private
-    ).freeze
-
-    attr_reader :allowed_updates,
-                :credentials,
-                :dependencies,
-                :existing_pull_requests,
-                :existing_group_pull_requests,
-                :id,
-                :ignore_conditions,
-                :package_manager,
-                :requirements_update_strategy,
-                :security_advisories,
-                :security_updates_only,
-                :source,
-                :token,
-                :vendor_dependencies,
-                :dependency_groups,
-                :dependency_group_to_refresh
+    ).freeze, T::Array[Symbol])
+
+    sig { returns(T::Array[T::Hash[String, T.untyped]]) }
+    attr_reader :allowed_updates
+
+    sig { returns(T::Array[T::Hash[String, T.any(T::Boolean, String)]]) }
+    attr_reader :credentials
+
+    sig { returns(T.nilable(T::Array[String])) }
+    attr_reader :dependencies
+
+    sig { returns(T::Array[T.untyped]) }
```

Comment:
Should this be something like 
```suggestion
    sig { returns(T::Array[T::Hash[T::Symbol, T.any(T::String, T::Bool)]]) }
```

Reply:
On the one hand it seems tedious to type all of that out and read it, but on the other hand, I suppose it's just the price we pay for the benefits of having types

Reply:
I mentioned above in response to @landongrindheim I think it would be good to construct classes here instead of leaving them as raw arrays of hashes. `ExistingPullRequest` doesn't exist yet but it should! I think it would bring much needed clarity to the codebase.

Reply:
I wonder if it would be a better approach to leave this file as untyped, but then slowly type each of the attrs with the actual backing class?

Reply:
Maybe `T::Array[T::Hash[T::Symbol, T.untyped]]` as a middle ground?

Reply:
Sure, but I think `existing_pull_requests` is actually a `T::Array[T::Array[T::Hash[String, T.untyped]]]` 

https://github.com/dependabot/dependabot-core/blob/f6f2548580dab0ad1b5b6fca2568c14470e6e157/updater/spec/dependabot/updater_spec.rb#L868-L875

Reply:
Ah apologies your comments hadn't yet loaded when I suggested that, I think your suggestion of adding types to the backing class is good!